### PR TITLE
Merge Develop to Master

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**

### CODE REVIEW

**Comment by WilliamXieMSFT** [comment]
In CodeFlow, Iteration 8-9 is the Copyright Header fix. I recommend viewing iteration 1-8 and 9-18 comparisons to avoid the 500+ files of copyright header changes.

File: src/InstrumentationEngine/ProfilerManager.cpp [hunk 1]
```diff
@@ -139,15 +139,23 @@ HRESULT MicrosoftInstrumentationEngine::CProfilerManager::SetupProfilingEnvironm
     // to co create a free threaded version of msxml on a thread that it owns to avoid this.
     CHandle hConfigThread(CreateThread(NULL, 0, InstrumentationMethodThreadProc, this, 0, NULL));
 
-    DWORD retVal = WaitForSingleObject(hConfigThread, 15000);
-    if (retVal == WAIT_ABANDONED)
+    DWORD waitTime = 60 * 1000; // Wait 1 minute for loading instrumentation methods
+#ifdef Debug
```
Comment (wiktork) [inline src/InstrumentationEngine/ProfilerManager.cpp]: I'm wondering if we need to parameterize this at some point.

Comment (WilliamXieMSFT) [reply src/InstrumentationEngine/ProfilerManager.cpp]: +1, Definitely something to consider for the next iteration

---
In reply to: [308869827](https://github.com/microsoft/CLRInstrumentationEngine/pull/111#discussion_r308869827) [](ancestors = 308869827)

**Comment by wiktork** [comment]
>     // TODO: Investigate why call to CoUninitialize deadlocks the thread

any issues here?

---
Refers to: src/Common.Lib/EventLoggingBase.cpp:133 in b0a8b26. [](commit_id = b0a8b269999fc552f8a7a570caeb90006a0c659d, deletion_comment = False)

**Comment by wiktork** [comment]
>     // TODO: Investigate why call to CoUninitialize deadlocks the thread

nvm i see it got fixed laster on.

---
In reply to: [516536183](https://github.com/microsoft/CLRInstrumentationEngine/pull/111#issuecomment-516536183) [](ancestors = 516536183)

---
Refers to: src/Common.Lib/EventLoggingBase.cpp:133 in cac2e27. [](commit_id = cac2e272e1a5f4d8046c653f9f418a4a90d853d4, deletion_comment = False)

**Comment by wiktork** [comment]
>     Shutdown();

Do we still shutdown somewhere else?

---
Refers to: src/InstrumentationEngine.Lib/LoggerService.cpp:28 in cac2e27. [](commit_id = cac2e272e1a5f4d8046c653f9f418a4a90d853d4, deletion_comment = True)

**Comment by WilliamXieMSFT** [comment]
>     // TODO: Investigate why call to CoUninitialize deadlocks the thread

Yep, this was due to the IM's holding a static global pointer

---
In reply to: [516536544](https://github.com/microsoft/CLRInstrumentationEngine/pull/111#issuecomment-516536544) [](ancestors = 516536544,516536183)

---
Refers to: src/Common.Lib/EventLoggingBase.cpp:133 in b0a8b26. [](commit_id = b0a8b269999fc552f8a7a570caeb90006a0c659d, deletion_comment = False)

**Comment by WilliamXieMSFT** [comment]
>     Shutdown();

We do on CLogging::Shutdown which refcounts the CLoggerService. All entrypoints are currently invoking CLogging::Shutdown correctly.

---
In reply to: [516537078](https://github.com/microsoft/CLRInstrumentationEngine/pull/111#issuecomment-516537078) [](ancestors = 516537078)

---
Refers to: src/InstrumentationEngine.Lib/LoggerService.cpp:28 in cac2e27. [](commit_id = cac2e272e1a5f4d8046c653f9f418a4a90d853d4, deletion_comment = True)

File: build/Tasks/ZipArchiveTask.cs [hunk 2]
```diff
@@ -41,6 +41,14 @@ public override bool Execute()
...
                     {
                         string destinationFileName = new FileInfo(fileItem.ItemSpec).Name;
+
+                        // Prevent duplicate files from getting zipped.
+                        if (uniqueFiles.Contains(destinationFileName))
+                        {
+                            continue;
+                        }
+
+                        uniqueFiles.Add(destinationFileName);
```
Comment (WilliamXieMSFT) [inline build/Tasks/ZipArchiveTask.cs]: Merge conflict reintroduced the same code twice...

**Pull request merged**